emscripten/WebARKitJS.cpp

```diff
@@ -1,45 +1,46 @@
 #include "WebARKitJS.h"
 
 void WebARKit::initTrackerGray(emscripten::val data_buffer, int width, int height) {
-  auto u8 =
-      emscripten::convertJSArrayToNumberVector<uint8_t>(data_buffer);
-  manager.initTracker(u8.data(), width, height);
+    auto u8 = emscripten::convertJSArrayToNumberVector<uint8_t>(data_buffer);
+    manager.initTracker(u8.data(), width, height);
 }
 
 void WebARKit::processFrame(emscripten::val data_buffer, webarkit::ColorSpace colorSpace) {
-  auto u8 =
-      emscripten::convertJSArrayToNumberVector<uint8_t>(data_buffer);
-  manager.processFrameData(u8.data(), this->videoWidth,
-                           this->videoHeight, colorSpace);
+    auto u8 = emscripten::convertJSArrayToNumberVector<uint8_t>(data_buffer);
+    manager.processFrameData(u8.data(), this->videoWidth, this->videoHeight, colorSpace);
 }
 
-void WebARKit::setLogLevel(int logLevel) {
-  manager.setLogLevel(logLevel);
-}
+void WebARKit::setLogLevel(int logLevel) { manager.setLogLevel(logLevel); }
 
 emscripten::val WebARKit::getHomography() {
-  std::vector<double> output;
-   output = manager.getOutputData();
-  emscripten::val homography = emscripten::val::array();
-  for (auto i = 0; i < 9; i++) {
-    homography.call<void>("push", output[i]);
-  }
-  return homography;
+    std::vector<double> output;
+    output = manager.getOutputData();
+    emscripten::val homography = emscripten::val::array();
+    for (auto i = 0; i < 9; i++) {
+        homography.call<void>("push", output[i]);
+    }
+    return homography;
 }
 
+emscripten::val WebARKit::getPoseMatrix() {
+    cv::Mat poseMatrix = manager.getPoseMatrix();
+    emscripten::val pose = emscripten::val::array();
+    for (auto i = 0; i < 12; i++) {
+        pose.call<void>("push", poseMatrix.at<double>(i));
```

Comment:
This seems causing hte issue in webarkit-testing see [comment](https://github.com/webarkit/WebARKitLib/pull/23#issuecomment-1791254849).
If i replace this line with:

```diff
--pose.call<void>("push", poseMatrix.at<double>(i));
++pose.call<void>("push", 0.0);
```

i got no errors while tracking (of curse it is dipslying a set of zeros...). This suggest me that this code or related code need to be changed

Reply:
thsi was solved with commit https://github.com/kalwalt/webarkit-testing/pull/24/commits/64e4c78937228b66ae561da71c5ee36e829151f3